# Checkpoint 1

# BUSINESS UNDERSTANDING

## Business Objective

Proyek ini bertujuan untuk mengembangkan model prediksi harga laptop yang akurat berdasarkan spesifikasi. Model ini akan membantu konsumen, retailer, dan produsen dalam menetapkan harga yang adil, mengoptimalkan strategi harga, serta membuat keputusan berbasis data. Dengan memahami faktor-faktor yang mempengaruhi harga dan memanfaatkan analitik data, proyek ini akan menciptakan sistem prediksi yang transparan dan efisien di pasar laptop.

## Asses Situation

Situasi bisnis yang mendasari analisis ini adalah lingkungan pasar yang semakin kompetitif dan dinamis di industri teknologi, khususnya dalam penjualan laptop. Perkembangan teknologi, dan variasi spesifikasi produk mengharuskan perusahaan untuk memiliki pemahaman yang mendalam tentang faktor-faktor yang memengaruhi harga laptop. Pengembangan model prediksi harga berbasis spesifikasi menjadi penting untuk meningkatkan efisiensi operasional, membantu strategi penetapan harga, dan memberikan rekomendasi yang lebih akurat kepada pelanggan. Oleh karena itu, analisis prediksi harga laptop berdasarkan spesifikasi menjadi krusial dalam mengidentifikasi peluang dan tantangan dalam menghadapi pasar teknologi yang terus berkembang.

## Data Mining Goals

Dengan mencari informasi tentang spesifikasi laptop, dapat membantu mengidentifikasi dan menganalisis faktor-faktor yang mempengaruhi harga laptop berdasarkan berbagai aspek seperti merek, spesifikasi teknis, perangkat lunak, serta tambahan aksesori dan desain. Selain itu, analisis dapat mengungkap sejauh mana faktor estetika dan merek dapat menentukan suatu harga laptop.

## Project Plan

Proyek ini akan dimulai dengan penelitian mengenai faktor-faktor yang memengaruhi harga laptop, seperti merek, spesifikasi teknis, perangkat lunak bawaan, aksesori tambahan, dan desain. Tahap awal mencakup pengumpulan data dari berbagai sumber, termasuk e-commerce dan situs resmi produsen laptop. Setelah data terkumpul, analisis eksploratif akan dilakukan untuk mengidentifikasi tren harga serta faktor dominan yang berpengaruh. Selain itu, proyek ini juga akan mengevaluasi dampak paket pembelian terhadap harga laptop, termasuk apakah keberadaan aksesori seperti monitor, keyboard, mouse, atau printer memberikan nilai tambah yang signifikan. Pada tahap akhir, hasil penelitian akan dirangkum dalam laporan analitis yang dilengkapi dengan visualisasi data, memberikan wawasan mendalam mengenai strategi penetapan harga di industri laptop.

# Checkpoint 2

# DATA UNDERSTANDING

Link Dataset https://www.kaggle.com/datasets/arnabchaki/laptop-price-prediction

In [16]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("laptops_test.csv")

df

Manufacturer                           Model Name            Category  \
0             HP    15-bs053od (i7-7500U/6GB/1TB/W10)            Notebook   
1           Asus                     Rog GL753VE-DS74              Gaming   
2           Dell                        Inspiron 7579  2 in 1 Convertible   
3        Toshiba                    Portege Z30-C-1CV            Notebook   
4         Lenovo                    IdeaPad 320-15ABR            Notebook   
..           ...                                  ...                 ...   
320       Lenovo                       Yoga 500-14ISK  2 in 1 Convertible   
321       Lenovo                       Yoga 900-13ISK  2 in 1 Convertible   
322       Lenovo                   IdeaPad 100S-14IBR            Notebook   
323           HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook   
324         Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook   

    Screen Size                                      Screen  \
0         15.6"                                    1366x768   
1         17.3"                           Full HD 1920x1080   
2         15.6"   IPS Panel Full HD / Touchscreen 1920x1080   
3         13.3"                           Full HD 1920x1080   
4         15.6"                           Full HD 1920x1080   
..          ...                                         ...   
320       14.0"   IPS Panel Full HD / Touchscreen 1920x1080   
321       13.3"  IPS Panel Quad HD+ / Touchscreen 3200x1800   
322       14.0"                                    1366x768   
323       15.6"                                    1366x768   
324       15.6"                                    1366x768   

                                      CPU   RAM               Storage  \
0              Intel Core i7 7500U 2.7GHz   6GB               1TB HDD   
1             Intel Core i7 7700HQ 2.8GHz  16GB  256GB SSD +  1TB HDD   
2              Intel Core i7 7500U 2.7GHz  12GB             512GB SSD   
3              Intel Core i5 6200U 2.3GHz   4GB             128GB SSD   
4             AMD A12-Series 9720P 3.6GHz   6GB             256GB SSD   
..                                    ...   ...                   ...   
320            Intel Core i7 6500U 2.5GHz   4GB             128GB SSD   
321            Intel Core i7 6500U 2.5GHz  16GB             512GB SSD   
322  Intel Celeron Dual Core N3050 1.6GHz   2GB    64GB Flash Storage   
323            Intel Core i7 6500U 2.5GHz   6GB               1TB HDD   
324  Intel Celeron Dual Core N3050 1.6GHz   4GB             500GB HDD   

                            GPU Operating System Operating System Version  \
0         Intel HD Graphics 620          Windows                       10   
1    Nvidia GeForce GTX 1050 Ti          Windows                       10   
2         Intel HD Graphics 620          Windows                       10   
3         Intel HD Graphics 520          Windows                        7   
4                AMD Radeon 530          Windows                       10   
..                          ...              ...                      ...   
320       Intel HD Graphics 520          Windows                       10   
321       Intel HD Graphics 520          Windows                       10   
322           Intel HD Graphics          Windows                       10   
323          AMD Radeon R5 M330          Windows                       10   
324           Intel HD Graphics          Windows                       10   

     Weight       Price  
0    2.04kg   5148468.0  
1    2.99kg  15552108.0  
2    2.19kg  11550708.0  
3     1.2kg  10625940.0  
4     2.2kg   4881708.0  
..      ...         ...  
320   1.8kg   5673096.0  
321   1.3kg  13329108.0  
322   1.5kg   2036268.0  
323  2.19kg   6793488.0  
324   2.2kg   3281148.0  

[325 rows x 13 columns]

In [4]:
print(f"Dataset memiliki {df.shape[0]} baris dan {df.shape[1]} kolom.\n")

Dataset memiliki 325 baris dan 13 kolom.



Disini saya menampilkan jumllah dataset yang masih belum dianalisis, berjumlah 325 baris dan 13 kolom.

# Cek tipe data setiap kolom

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 325 entries, 0 to 324
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Manufacturer              325 non-null    object 
 1   Model Name                325 non-null    object 
 2   Category                  325 non-null    object 
 3   Screen Size               325 non-null    object 
 4   Screen                    325 non-null    object 
 5   CPU                       325 non-null    object 
 6   RAM                       325 non-null    object 
 7    Storage                  325 non-null    object 
 8   GPU                       325 non-null    object 
 9   Operating System          325 non-null    object 
 10  Operating System Version  291 non-null    object 
 11  Weight                    325 non-null    object 
 12  Price                     325 non-null    float64
dtypes: float64(1), object(12)
memory usage: 33.1+ KB


In [6]:
df.describe()

Price
count  3.250000e+02
mean   9.891671e+06
std    5.943217e+06
min    1.547208e+06
25%    5.317416e+06
50%    8.892000e+06
75%    1.332022e+07
max    3.534570e+07

# Data cleaning

**1. Missing Value**

Selanjutnya saya mencari mising value dari dataset ini.

In [7]:
print((df.isna().sum() / len(df)) * 100)

Manufacturer                 0.000000
Model Name                   0.000000
Category                     0.000000
Screen Size                  0.000000
Screen                       0.000000
CPU                          0.000000
RAM                          0.000000
 Storage                     0.000000
GPU                          0.000000
Operating System             0.000000
Operating System Version    10.461538
Weight                       0.000000
Price                        0.000000
dtype: float64


di kolom operating system version terdapat 10% data yang hilang.

In [8]:
df[df['Operating System Version'].isnull()]

Manufacturer                             Model Name            Category  \
14        Lenovo         V110-15IAP (N3350/4GB/128GB/No            Notebook   
71          Asus                      Chromebook C202SA             Netbook   
78            HP                                 250 G5            Notebook   
83          Asus                      Rog G752VL-GC088D              Gaming   
110           HP                          Chromebook 13            Notebook   
112         Dell                          Inspiron 3552            Notebook   
113           HP                                 250 G6            Notebook   
124         Acer                          Chromebook 15            Notebook   
136       Lenovo                              Yoga Book  2 in 1 Convertible   
145         Dell                         Latitude E5570            Notebook   
150       Lenovo                     Legion Y520-15IKBN              Gaming   
172       Lenovo        V110-15ISK (i3-6006U/4GB/1TB/No            Notebook   
174       Lenovo        B51-80 (i5-6200U/8GB/1TB/Radeon            Notebook   
186           HP                                 250 G4            Notebook   
195       Lenovo      V110-15ISK (i5-6200U/4GB/500GB/No            Notebook   
196       Lenovo                      IdeaPad 510-15IKB            Notebook   
204         Dell                            Vostro 5568            Notebook   
210         Dell                                 XPS 13           Ultrabook   
213      Samsung                        Chromebook Plus  2 in 1 Convertible   
218       Lenovo    V110-15ISK (i3-6006U/4GB/1TB/Radeon            Notebook   
220         Acer                               Aspire 3            Notebook   
223       Lenovo                      IdeaPad 310-15ISK            Notebook   
237         Acer       C740-C9QX (3205U/2GB/32GB/Chrome             Netbook   
239       Lenovo                      IdeaPad 320-17IKB            Notebook   
249         Dell                            Vostro 3568            Notebook   
264         Acer                         Aspire ES1-572            Notebook   
270       Lenovo  V510-15IKB (i5-7200U/8GB/256GB/FHD/No            Notebook   
272         Dell                          Inspiron 3552            Notebook   
285         Acer                         Aspire ES1-531            Notebook   
289         Dell                          Inspiron 3567            Notebook   
299         Acer                         Aspire ES1-531            Notebook   
303         Dell                          Inspiron 3567            Notebook   
313         Acer                         Aspire ES1-531            Notebook   
317         Dell                          Inspiron 3567            Notebook   

    Screen Size                            Screen  \
14        15.6"                          1366x768   
71        11.6"                          1366x768   
78        15.6"                          1366x768   
83        17.3"       IPS Panel Full HD 1920x1080   
110       13.3"                Quad HD+ 3200x1800   
112       15.6"                          1366x768   
113       15.6"                          1366x768   
124       15.6"                          1366x768   
136       10.1"   IPS Panel Touchscreen 1920x1200   
145       15.6"                          1366x768   
150       15.6"       IPS Panel Full HD 1920x1080   
172       15.6"                          1366x768   
174       15.6"                 Full HD 1920x1080   
186       15.6"                          1366x768   
195       15.6"                          1366x768   
196       15.6"                 Full HD 1920x1080   
204       15.6"                 Full HD 1920x1080   
210       13.3"  Quad HD+ / Touchscreen 3200x1800   
213       12.3"   IPS Panel Touchscreen 2400x1600   
218       15.6"                          1366x768   
220       15.6"                          1366x768   
223       15.6"                          1366x768   
237       11.6"    

In [63]:
df['Operating System Version'].dropna().describe()

count            297
unique             5
top       Windows 11
freq              69
Name: Operating System Version, dtype: object

**2. Duplicated Values**

Selanjutnnya saya melakukan analilsis data yang duplikat

In [10]:
df[df.duplicated()]

Manufacturer                                 Model Name  \
297         Asus                       ZenBook UX305CA-UBM1   
298         Dell                              Inspiron 3567   
299         Acer                             Aspire ES1-531   
300         Dell                              Inspiron 3552   
301       Lenovo                         IdeaPad Y700-15ISK   
302           HP                        Pavilion 15-AW003nv   
303         Dell                              Inspiron 3567   
304           HP                           Stream 11-Y000na   
305         Asus  X556UJ-XO044T (i7-6500U/4GB/500GB/GeForce   
306       Lenovo                             Yoga 500-14ISK   
307       Lenovo                             Yoga 900-13ISK   
308       Lenovo                         IdeaPad 100S-14IBR   
309           HP        15-AC110nv (i7-6500U/6GB/1TB/Radeon   
310         Asus        X553SA-XX031T (N3050/4GB/500GB/W10)   
311         Asus                       ZenBook UX305CA-UBM1   
312         Dell                              Inspiron 3567   
313         Acer                             Aspire ES1-531   
314         Dell                              Inspiron 3552   
315       Lenovo                         IdeaPad Y700-15ISK   
316           HP                        Pavilion 15-AW003nv   
317         Dell                              Inspiron 3567   
318           HP                           Stream 11-Y000na   
319         Asus  X556UJ-XO044T (i7-6500U/4GB/500GB/GeForce   
320       Lenovo                             Yoga 500-14ISK   
321       Lenovo                             Yoga 900-13ISK   
322       Lenovo                         IdeaPad 100S-14IBR   
323           HP        15-AC110nv (i7-6500U/6GB/1TB/Radeon   
324         Asus        X553SA-XX031T (N3050/4GB/500GB/W10)   

               Category Screen Size  \
297           Ultrabook       13.3"   
298            Notebook       15.6"   
299            Notebook       15.6"   
300            Notebook       15.6"   
301            Notebook       15.6"   
302            Notebook       15.6"   
303            Notebook       15.6"   
304             Netbook       11.6"   
305            Notebook       15.6"   
306  2 in 1 Convertible       14.0"   
307  2 in 1 Convertible       13.3"   
308            Notebook       14.0"   
309            Notebook       15.6"   
310            Notebook       15.6"   
311           Ultrabook       13.3"   
312            Notebook       15.6"   
313            Notebook       15.6"   
314            Notebook       15.6"   
315            Notebook       15.6"   
316            Notebook       15.6"   
317            Notebook       15.6"   
318             Netbook       11.6"   
319            Notebook       15.6"   
320  2 in 1 Convertible       14.0"   
321  2 in 1 Convertible       13.3"   
322            Notebook       14.0"   
323            Notebook       15.6"   
324            Notebook       15.6"   

                                         Screen  \
297                 IPS Panel Full HD 1920x1080   
298                                    1366x768   
299                                    1366x768   
300                                    1366x768   
301                 IPS Panel Full HD 1920x1080   
302                           Full HD 1920x1080   
303                                    1366x768   
304                                    1366x768   
305                                    1366x768   
306   IPS Panel Full HD / Touchscreen 1920x1080   
307  IPS Panel Quad HD+ / Touchscreen 3200x1800   
308                                    1366x768   
309                                    1366x768   
310                                    1366x768   
311                 IPS Panel Full HD 1920x1080   
312                                    1366x768   
313                                    1366x768   
314                                    1366x768   
315                 IPS Panel Full HD 1920x1080   
316                     

didalam dataset ini ditemukan sebanyak 28 data duplikat

In [11]:
df = df.drop_duplicates()

selanjutnya data yang duplikat saya hapus

In [17]:
df

Manufacturer                           Model Name            Category  \
0             HP    15-bs053od (i7-7500U/6GB/1TB/W10)            Notebook   
1           Asus                     Rog GL753VE-DS74              Gaming   
2           Dell                        Inspiron 7579  2 in 1 Convertible   
3        Toshiba                    Portege Z30-C-1CV            Notebook   
4         Lenovo                    IdeaPad 320-15ABR            Notebook   
..           ...                                  ...                 ...   
292       Lenovo                       Yoga 500-14ISK  2 in 1 Convertible   
293       Lenovo                       Yoga 900-13ISK  2 in 1 Convertible   
294       Lenovo                   IdeaPad 100S-14IBR            Notebook   
295           HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook   
296         Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook   

    Screen Size                                      Screen  \
0         15.6"                                    1366x768   
1         17.3"                           Full HD 1920x1080   
2         15.6"   IPS Panel Full HD / Touchscreen 1920x1080   
3         13.3"                           Full HD 1920x1080   
4         15.6"                           Full HD 1920x1080   
..          ...                                         ...   
292       14.0"   IPS Panel Full HD / Touchscreen 1920x1080   
293       13.3"  IPS Panel Quad HD+ / Touchscreen 3200x1800   
294       14.0"                                    1366x768   
295       15.6"                                    1366x768   
296       15.6"                                    1366x768   

                                      CPU   RAM               Storage  \
0              Intel Core i7 7500U 2.7GHz   6GB               1TB HDD   
1             Intel Core i7 7700HQ 2.8GHz  16GB  256GB SSD +  1TB HDD   
2              Intel Core i7 7500U 2.7GHz  12GB             512GB SSD   
3              Intel Core i5 6200U 2.3GHz   4GB             128GB SSD   
4             AMD A12-Series 9720P 3.6GHz   6GB             256GB SSD   
..                                    ...   ...                   ...   
292            Intel Core i7 6500U 2.5GHz   4GB             128GB SSD   
293            Intel Core i7 6500U 2.5GHz  16GB             512GB SSD   
294  Intel Celeron Dual Core N3050 1.6GHz   2GB    64GB Flash Storage   
295            Intel Core i7 6500U 2.5GHz   6GB               1TB HDD   
296  Intel Celeron Dual Core N3050 1.6GHz   4GB             500GB HDD   

                            GPU Operating System Operating System Version  \
0         Intel HD Graphics 620          Windows                       10   
1    Nvidia GeForce GTX 1050 Ti          Windows                       10   
2         Intel HD Graphics 620          Windows                       10   
3         Intel HD Graphics 520          Windows                        7   
4                AMD Radeon 530          Windows                       10   
..                          ...              ...                      ...   
292       Intel HD Graphics 520          Windows                       10   
293       Intel HD Graphics 520          Windows                       10   
294           Intel HD Graphics          Windows                       10   
295          AMD Radeon R5 M330          Windows                       10   
296           Intel HD Graphics          Windows                       10   

     Weight       Price  
0    2.04kg   5148468.0  
1    2.99kg  15552108.0  
2    2.19kg  11550708.0  
3     1.2kg  10625940.0  
4     2.2kg   4881708.0  
..      ...         ...  
292   1.8kg   5673096.0  
293   1.3kg  13329108.0  
294   1.5kg   2036268.0  
295  2.19kg   6793488.0  
296   2.2kg   3281148.0  

[297 rows x 13 columns]

# Melakukan Imputasi

disini saya mengganti nilai yang hilang padakolom operating system version, dengan menyesuaikan os pada laptop yang ada, saya mengguanakan metode ini karena lebih efektif dibanding menggunakan mean karena jenis os yang berbeda, dan tidak saya hapus karena jumlah data yang hilang cukup banhyak

In [64]:
df.loc[df["Operating System"] == "Windows", "Operating System Version"] = df.loc[df["Operating System"] == "Windows", "Operating System Version"].fillna("Windows 10")
df.loc[df["Operating System"] == "macOS", "Operating System Version"] = df.loc[df["Operating System"] == "macOS", "Operating System Version"].fillna("macOS Monterey")
df.loc[df["Operating System"] == "Linux", "Operating System Version"] = df.loc[df["Operating System"] == "Linux", "Operating System Version"].fillna("Ubuntu 22.04")

In [67]:
df[df['Operating System Version'].isnull()]

Empty DataFrame
Columns: [Manufacturer, Model Name, Category, Screen Size, Screen, CPU, RAM, Storage, GPU, Operating System, Operating System Version, Weight, Price, Device Type]
Index: []

In [68]:
df

Manufacturer    Model Name     Category  Screen Size Screen          CPU  \
0           Dell  ROG Zephyrus     Business         16.0    IPS     Intel i7   
1           Asus  Spectre x360  Convertible         17.3     TN     Intel i7   
2           Acer  Spectre x360  Convertible         14.0   OLED  AMD Ryzen 5   
3          Apple        XPS 13    Ultrabook         16.0     TN     Intel i5   
4           Acer  ROG Zephyrus     Business         16.0     TN     Intel i5   
..           ...           ...          ...          ...    ...          ...   
292         Dell   ThinkPad X1    Ultrabook         14.0   OLED     Intel i9   
293        Apple       Swift 3       Gaming         14.0     TN  AMD Ryzen 5   
294         Acer       Swift 3       Gaming         15.6     TN     Intel i5   
295         Dell  ROG Zephyrus       Gaming         13.3     TN  AMD Ryzen 5   
296         Dell   MacBook Pro       Gaming         14.0     TN  AMD Ryzen 7   

      RAM    Storage               GPU Operating System  \
0    32GB    2TB SSD   NVIDIA RTX 3060            Linux   
1     8GB  512GB SSD   NVIDIA RTX 3050            Linux   
2    16GB    1TB SSD        Integrated            Linux   
3    32GB  512GB SSD   NVIDIA RTX 3050          Windows   
4    16GB  512GB SSD   NVIDIA RTX 3060          Windows   
..    ...        ...               ...              ...   
292   8GB    2TB SSD  AMD Radeon 6600M            macOS   
293  32GB    1TB SSD   NVIDIA RTX 3060            Linux   
294   8GB    1TB SSD   NVIDIA RTX 3060            macOS   
295  16GB  256GB SSD        Integrated            macOS   
296  16GB  512GB SSD  AMD Radeon 6600M          Windows   

    Operating System Version  Weight  Price Device Type  
0             macOS Monterey    3.04   1874      Laptop  
1                 Windows 10    2.98   3055      Laptop  
2                 Windows 11    1.47   2638      Laptop  
3                 Windows 11    1.58   2525      Laptop  
4                 Windows 11    3.29   2935      Laptop  
..                       ...     ...    ...         ...  
292               Windows 11    2.12   1013      Laptop  
293           macOS Monterey    3.17   1527      Laptop  
294               Windows 11    3.20   2259      Laptop  
295             Ubuntu 22.04    2.66   2363      Laptop  
296             Ubuntu 22.04    2.37    734      Laptop  

[297 rows x 14 columns]

In [69]:
print((df.isna().sum() / len(df)) * 100)

Manufacturer                0.0
Model Name                  0.0
Category                    0.0
Screen Size                 0.0
Screen                      0.0
CPU                         0.0
RAM                         0.0
Storage                     0.0
GPU                         0.0
Operating System            0.0
Operating System Version    0.0
Weight                      0.0
Price                       0.0
Device Type                 0.0
dtype: float64


sekarang sudah tidak ada datset sudah lengkap

**3. Outliers**

In [62]:
numerical_columns = ["Price", "Weight", "Screen Size"]

for col in numerical_columns:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1

    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]

    print(f"Jumlah outliers di '{col}': {len(outliers)}")
    print(outliers[[col]])

Jumlah outliers di 'Price': 0
Empty DataFrame
Columns: [Price]
Index: []
Jumlah outliers di 'Weight': 0
Empty DataFrame
Columns: [Weight]
Index: []
Jumlah outliers di 'Screen Size': 0
Empty DataFrame
Columns: [Screen Size]
Index: []


# Incosistent value

In [25]:
df

Manufacturer                           Model Name            Category  \
0             HP    15-bs053od (i7-7500U/6GB/1TB/W10)            Notebook   
1           Asus                     Rog GL753VE-DS74              Gaming   
2           Dell                        Inspiron 7579  2 in 1 Convertible   
3        Toshiba                    Portege Z30-C-1CV            Notebook   
4         Lenovo                    IdeaPad 320-15ABR            Notebook   
..           ...                                  ...                 ...   
292       Lenovo                       Yoga 500-14ISK  2 in 1 Convertible   
293       Lenovo                       Yoga 900-13ISK  2 in 1 Convertible   
294       Lenovo                   IdeaPad 100S-14IBR            Notebook   
295           HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook   
296         Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook   

    Screen Size                                      Screen  \
0         15.6"                                    1366x768   
1         17.3"                           Full HD 1920x1080   
2         15.6"   IPS Panel Full HD / Touchscreen 1920x1080   
3         13.3"                           Full HD 1920x1080   
4         15.6"                           Full HD 1920x1080   
..          ...                                         ...   
292       14.0"   IPS Panel Full HD / Touchscreen 1920x1080   
293       13.3"  IPS Panel Quad HD+ / Touchscreen 3200x1800   
294       14.0"                                    1366x768   
295       15.6"                                    1366x768   
296       15.6"                                    1366x768   

                                      CPU   RAM               Storage  \
0              Intel Core i7 7500U 2.7GHz   6GB               1TB HDD   
1             Intel Core i7 7700HQ 2.8GHz  16GB  256GB SSD +  1TB HDD   
2              Intel Core i7 7500U 2.7GHz  12GB             512GB SSD   
3              Intel Core i5 6200U 2.3GHz   4GB             128GB SSD   
4             AMD A12-Series 9720P 3.6GHz   6GB             256GB SSD   
..                                    ...   ...                   ...   
292            Intel Core i7 6500U 2.5GHz   4GB             128GB SSD   
293            Intel Core i7 6500U 2.5GHz  16GB             512GB SSD   
294  Intel Celeron Dual Core N3050 1.6GHz   2GB    64GB Flash Storage   
295            Intel Core i7 6500U 2.5GHz   6GB               1TB HDD   
296  Intel Celeron Dual Core N3050 1.6GHz   4GB             500GB HDD   

                            GPU Operating System Operating System Version  \
0         Intel HD Graphics 620          Windows                       10   
1    Nvidia GeForce GTX 1050 Ti          Windows                       10   
2         Intel HD Graphics 620          Windows                       10   
3         Intel HD Graphics 520          Windows                        7   
4                AMD Radeon 530          Windows                       10   
..                          ...              ...                      ...   
292       Intel HD Graphics 520          Windows                       10   
293       Intel HD Graphics 520          Windows                       10   
294           Intel HD Graphics          Windows                       10   
295          AMD Radeon R5 M330          Windows                       10   
296           Intel HD Graphics          Windows                       10   

     Weight       Price  
0    2.04kg   5148468.0  
1    2.99kg  15552108.0  
2    2.19kg  11550708.0  
3     1.2kg  10625940.0  
4     2.2kg   4881708.0  
..      ...         ...  
292   1.8kg   5673096.0  
293   1.3kg  13329108.0  
294   1.5kg   2036268.0  
295  2.19kg   6793488.0  
296   2.2kg   3281148.0  

[297 rows x 13 columns]

Menuryt saya dataset ini cukup konsisten

# Construct Data

In [51]:
def classify_device(row):
    if row["Category"] in ["Ultrabook", "Gaming", "Business", "Convertible"]:
        return "Laptop"
    elif row["Screen Size"] > 17.3 or row["Weight"] > 3.5:
        return "Desktop"

df["Device Type"] = df.apply(classify_device, axis=1)

print(df.head())

  Manufacturer    Model Name     Category  Screen Size Screen          CPU  \
0         Dell  ROG Zephyrus     Business         16.0    IPS     Intel i7   
1         Asus  Spectre x360  Convertible         17.3     TN     Intel i7   
2         Acer  Spectre x360  Convertible         14.0   OLED  AMD Ryzen 5   
3        Apple        XPS 13    Ultrabook         16.0     TN     Intel i5   
4         Acer  ROG Zephyrus     Business         16.0     TN     Intel i5   

    RAM    Storage              GPU Operating System Operating System Version  \
0  32GB    2TB SSD  NVIDIA RTX 3060            Linux           macOS Monterey   
1   8GB  512GB SSD  NVIDIA RTX 3050            Linux               Windows 10   
2  16GB    1TB SSD       Integrated            Linux               Windows 11   
3  32GB  512GB SSD  NVIDIA RTX 3050          Windows               Windows 11   
4  16GB  512GB SSD  NVIDIA RTX 3060          Windows               Windows 11   

   Weight  Price Device Type  
0    3.04   1

disini saya menambahkan kolom device type untuk membedakan laptop dan desktop

# Data Reduction

In [70]:
df_desktop = df[df["Device Type"] == "Desktop"]
print(df_desktop)

Empty DataFrame
Columns: [Manufacturer, Model Name, Category, Screen Size, Screen, CPU, RAM, Storage, GPU, Operating System, Operating System Version, Weight, Price, Device Type]
Index: []


Karena kolom yang saya buat tadi tidak cukup berguna dan hanya memboros kolom maka  kolom device type saya hapus

In [71]:
del df["Device Type"]

In [72]:
df

Manufacturer    Model Name     Category  Screen Size Screen          CPU  \
0           Dell  ROG Zephyrus     Business         16.0    IPS     Intel i7   
1           Asus  Spectre x360  Convertible         17.3     TN     Intel i7   
2           Acer  Spectre x360  Convertible         14.0   OLED  AMD Ryzen 5   
3          Apple        XPS 13    Ultrabook         16.0     TN     Intel i5   
4           Acer  ROG Zephyrus     Business         16.0     TN     Intel i5   
..           ...           ...          ...          ...    ...          ...   
292         Dell   ThinkPad X1    Ultrabook         14.0   OLED     Intel i9   
293        Apple       Swift 3       Gaming         14.0     TN  AMD Ryzen 5   
294         Acer       Swift 3       Gaming         15.6     TN     Intel i5   
295         Dell  ROG Zephyrus       Gaming         13.3     TN  AMD Ryzen 5   
296         Dell   MacBook Pro       Gaming         14.0     TN  AMD Ryzen 7   

      RAM    Storage               GPU Operating System  \
0    32GB    2TB SSD   NVIDIA RTX 3060            Linux   
1     8GB  512GB SSD   NVIDIA RTX 3050            Linux   
2    16GB    1TB SSD        Integrated            Linux   
3    32GB  512GB SSD   NVIDIA RTX 3050          Windows   
4    16GB  512GB SSD   NVIDIA RTX 3060          Windows   
..    ...        ...               ...              ...   
292   8GB    2TB SSD  AMD Radeon 6600M            macOS   
293  32GB    1TB SSD   NVIDIA RTX 3060            Linux   
294   8GB    1TB SSD   NVIDIA RTX 3060            macOS   
295  16GB  256GB SSD        Integrated            macOS   
296  16GB  512GB SSD  AMD Radeon 6600M          Windows   

    Operating System Version  Weight  Price  
0             macOS Monterey    3.04   1874  
1                 Windows 10    2.98   3055  
2                 Windows 11    1.47   2638  
3                 Windows 11    1.58   2525  
4                 Windows 11    3.29   2935  
..                       ...     ...    ...  
292               Windows 11    2.12   1013  
293           macOS Monterey    3.17   1527  
294               Windows 11    3.20   2259  
295             Ubuntu 22.04    2.66   2363  
296             Ubuntu 22.04    2.37    734  

[297 rows x 13 columns]